2018-07-16-tips-from-the-tour-part-2-drift-analysis-and-more-integrations.md

```diff
@@ -0,0 +1,77 @@
+A SmartState analysis of the VM has been performed before and after the WordPress playbook run.
+Using the Drift Analysis feature we compared the VMs configurations and highlighted the differences on packages:
+
+Triangle icons in the picture highlight the packages that changed, as you can see the php-fpm package was missing and is now installed. Versions and releases are reported too.  
+SmartState analysis can be scheduled to run so we can track all the configuration changes of Linux and Windows VMs on their whole lifecycle.
+
+## Integration with Red Hat OpenShift Container Platform ##
+
+Adding OpenShift requires the rules and token to access API and get events from hawkular as reported here:
+[https://access.redhat.com/documentation/en-us/red_hat_cloudforms/4.6/html/managing_providers/containers_providers]
+  
+Since we were using one of the latest OpenShift Container Platform (OCP) releases all the required service account and roles required by Red Hat CloudForms are installed by default.
+We just had to retrieve the management-admin token on the console with the following command on the OCP master node:  
+  
+oc sa get-token -n management-infra management-admin
+  
+The token must be provided when the provider is added in the configuration screen both for the credentials and metric (Hawkular) tab.
+  
+The topology view of a OCP project is one of the coolest visual features and allows to have a view of our CI/CD demo project including Pods, Replication controllers, Services, Routes, Nodes and allows to have a 10k view of what’s going on in a project or on a OCP cluster.
+
+CloudForms also allows to aggregate metrics of pods showing cores, memory, network IO metrics to understand overall data related to projects and use the data as a base for the chargeback feature:
+
+We also demoed the ability to perform massive openSCAP analysis of container images to assess the status of our infrastructure.  
+  
+Container analysis works by analyzing the target container image starting a manageiq-scan pod in the namespace management-infra. The namespace is dedicated on OCP and doesn’t impact production. Here’s the output of the command oc get pods –all-namespaces while the analysis is running:
+
+When the scan is completed the pod is deleted and results are uploaded in the CloudForms DB. SmartState reports all the software packages and versions in the container image and OpenSCAP Failed Rules, drill down information is available and an extensive openSCAP HTML Report is available too:
+
+A custom CloudForms Report (see downloads above) has been created to show the overall compliance status of the whole IT environment and published on the dashboard:
+
+Thanks to [Nick Catling] [https://www.redhat.com/en/blog] for the help in preparing the demo.
+  
+Resources
+[https://cloudformsblog.redhat.com/2016/08/22/getting-started-with-managing-vmware-with-red-hat-cloudforms/]
```

Comment:
We're going to have to relink all of these references, but we can also do that in a follow up.